pkg/eval/vals/conversion.go

```diff
@@ -102,6 +102,11 @@ func ScanToGo(src any, ptr any) error {
 			return fmt.Errorf("internal bug: need pointer to scan to, got %T", ptr)
 		}
 		dstType := ptrType.Elem()
+		if dstType.String() == "eval.Callable" && ValueOf(src) == ValueOf(nil) {
```

Comment:
Hmm, I decided to keep the check against eval.Callable instead of against all interface types in general; this is because I don't want list and map arguments to admit `$nil` as valid values, since a lot of builtin functions that operate on them assume non-nil-ness. But for Callable values, I believe all builtin functions already assume `nil` as a possible value.

Reply:
Ha, now I get why you said this would create a cyclic dependency. This can be easily worked around by setting the value of `callableType` from the `eval` package instead.

Reply:
The import cycle of the test file can be more easily worked around by putting the test in a `vals_test` package, but I opted for a more hacky way of mocking out the `CallableType`.